File: p1/StopWatch.java
package problem1;

public class StopWatch {
	int flag = 0;
	
	public boolean start()
	{
		if(flag != 0)
			return false;
		
		//logic
		flag = 1;
		return true;
	}
	public boolean stop()
	{
		if(flag != 1 || flag!= 2)
			return false;
		
		//logic
		flag = 0;
		return true;
	}
	public boolean pause()
	{
		if(flag != 1)
			return false;
		
		//logic
		flag = 2;
		return true;
	}
	public boolean resume()
	{
		if(flag != 2)
			return false;
		
		//logic
		flag = 1;
		return true;
	}
}

File: p1/StopWatch.py
class StopWatch:
	
  def __init__(self):
    self.flag = 0
	
  def start(self):
    if self.flag != 0:
      return self.false
		
    #logic
    self.flag = 1
    return true
	
  def stop(self):
    if self.flag != 1 or self.flag!= 2 :
      return false
		
		#logic
    self.flag = 0
    return true
	
  def pause():
    if self.flag != 1:
      return self.false
		
		#logic
    self.flag = 2
    return true
	
  def resume(self):
    if self.flag != 2:
      return self.false
		
		#logic
    self.flag = 1
    return true
